spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/Tokenizer.java

```diff
@@ -0,0 +1,127 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.plugin.spark;
+
+import co.cask.cdap.api.annotation.Description;
+import co.cask.cdap.api.annotation.Name;
+import co.cask.cdap.api.annotation.Plugin;
+import co.cask.cdap.api.data.format.StructuredRecord;
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.plugin.PluginConfig;
+import co.cask.cdap.etl.api.PipelineConfigurer;
+import co.cask.cdap.etl.api.batch.SparkCompute;
+import co.cask.cdap.etl.api.batch.SparkExecutionPluginContext;
+import com.google.common.base.Preconditions;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.JavaSparkContext;
+import org.apache.spark.api.java.function.Function;
+import org.apache.spark.ml.feature.RegexTokenizer;
+import org.apache.spark.sql.DataFrame;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.RowFactory;
+import org.apache.spark.sql.SQLContext;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.Metadata;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+
+/**
+ * Tokenizer-SparkCompute that breaks text(such as sentence) into individual terms(usually words)
+ */
+@Plugin(type = SparkCompute.PLUGIN_TYPE)
+@Name(Tokenizer.PLUGIN_NAME)
+@Description("Used to tokenize text(such as sentence) into individual terms(usually words)")
+public class Tokenizer extends SparkCompute<StructuredRecord, StructuredRecord> {
+
+    public static final String PLUGIN_NAME = "Tokenizer";
+    private Config config;
+    public Schema outputSchema;
+
+    public Tokenizer(Config config) {
+        this.config = config;
+    }
+
+    /**
+     * Configuration for the Tokenizer Plugin.
+     */
+    public static class Config extends PluginConfig {
+        @Description("Column on which tokenization is to be done")
+        private final String columnToBeTokenized;
+
+        @Description("Delimiter for tokenization")
+        private final String delimiter;
+
+        @Description("Output column name for tokenized data")
+        private final String outputColumn;
+
+        public Config(String outputColumn, String columnToBeTokenized, String delimiter) {
+            this.columnToBeTokenized = columnToBeTokenized;
+            this.delimiter = delimiter;
+            this.outputColumn = outputColumn;
+        }
+    }
+
+    @Override
+    public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+        super.configurePipeline(pipelineConfigurer);
+        Schema inputSchema = pipelineConfigurer.getStageConfigurer().getInputSchema();
+        if (inputSchema != null) {
+            Schema.Type columnToBeTokenizedType = inputSchema.getField(config.columnToBeTokenized)
```

Comment:
we should also check for existence of the column in the schema. `inputSchema.getField` can return null.